### DISCUSSION TITLE
Why is the search result always 0 when calculating binary vectors using Hamming distance？

### DISCUSSION THREAD

**Discussion by yesletgo** [milvus-io/milvus#31969] [Q&A and General discussion]
![image](https://github.com/milvus-io/milvus/assets/21998052/ba4d4ce9-495f-4f27-af46-86e0730ef9a3)
![image](https://github.com/milvus-io/milvus/assets/21998052/0a5d2e4f-e520-4a9c-84b2-9b0eaa3ba31b)

![image](https://github.com/milvus-io/milvus/assets/21998052/35a43009-2989-49b2-854a-eac28e5bc10c)

**Comment by xiaofan-luan** [comment]
@yesletgo 
did you check IDs? 
maybe they are all [1,1,1,1....] so distance is all 0

**Reply by yesletgo** [reply]
Isn't the top k sorted by hummin distance? There are many in the library that are not [1,1,1...], so there should be some with scores other than 0

**Comment by yhmo** [comment]
Hamming distance:
0000 and 1111 is 4
1111 and 1111 is 0

Hamming distance = 0 means the two vectors are the same.

There are 500000000 vectors in the collection. Do you know how many [1,1,1...] in the collection? I guess there are millions of  [1,1,1...] in the collection.
The search result is sorted by hamming distance, from zero to non-zero. So, it returns the top 100 items with distance =0. The top 100 vectors are all [1,1,1...].
If you set topk=1000000, maybe there are some distances not 0, but it is impossible because milvus only accepts the max topk 16384.

**Reply by yesletgo** [reply]
Thank you. I mistakenly thought it was sorted in descending order based on the Hamming distance value, but currently it seems to be sorted in ascending order.
